### ISSUE TITLE
How is juno-uber installed? Is access to atom.io needed with the thin installer.

### ISSUE THREAD

**Issue by SteffenPL** [heetbeet/juliawin#44]
First of all, this package is great and very useful!

My collaborator is behind a firewall, and I will recommend him to use juliawin. 

I wanted to ask which URL's the thin installer accesses? Only GitHub files or also external URLs?
(Depending on that I would recommend my collaborator the full or thin installer).

PS: This is not really an issue. But I didn't know where to ask.

**Comment by heetbeet** [comment]
This is a good place to ask questions :)

The main approach is to first bootstrap as little as possible (basically just download Julia from the official website) and then within Julia run the rest of the installer. This includes downloading binaries directly from sources and then bootstrap them further by calling their own package managers etc.
Here is a list of some of the source that potentially get accesses by the thin wrapper:
- github.com (to download this repo, Atom and MinGW)
- julialang.org
- curl.haxx.se (for a windows curl implementation)
- code.visualstudio.com

Also whatever links to the Julia package manager.

And this is the routine to get all the atom packages:
```
apm install language-julia
apm install julia-client
apm install ink
apm install uber-juno
apm install latex-completions
apm install indent-detective
apm install hyperclick
apm install tool-bar
apm install file-watcher
```

**Comment by heetbeet** [comment]
What I would recommend is to run the thin-installer on another computer and then just copy the result over, since I don't quite know what all the updates were between the last release and now.

**Comment by heetbeet** [comment]
I want to make a new release, but am kind of waiting to do it after v1.6 is released.

**Comment by SteffenPL** [comment]
Thanks you for the fast reply!

If I understand correctly, than `apm install *` will probably access atom.io, since that's where the node.js packages are.
Is everything that gets installed portable? (Also git/mingw/atom)?

**Comment by heetbeet** [comment]
I would also guess that would come from `atom.io`.

**TL;DR** It is fully portable, and should work on a locked-down system, but some misbehaving programs might leave minor bit of trash behind.

The thin installer works as follows

1. Thin installer
- downloads this repo as a zip file to `%TEMP%`
- starts the locally downloaded installation .bat file: https://github.com/heetbeet/juliawin/blob/main/bin/Juliawin%20Setup.bat

2. `Juliawin Setup.bat` installer
- picks up from where the thin installer/initiator left off
- asks user to choose: installation location, programs to include with the installation, etc.
- downloads Julia and extracts to final destination
- start the locally downloaded Julia part of the installer: https://github.com/heetbeet/juliawin/blob/main/internals/juliawin_cli.jl

3. `juliawin_cli.jl` installer
- picks up from where `Juliawin Setup.bat` left off
- downloads all **portable** versions of sources required and extracts them to the correct locations
- installs and setup vscode extentions, atom extensions, OhMyREPL.jl and Revise.jl

Everything is wrapped with a thin layer of path/environment variable redirection to make things as portable as possible, and to allow user-data to end up in Juliawin's userdata directory.

Some minor issues still exists that will put off portable purists, but doesn't cause any administrative right to be violated and wouldn't clash with non-portable counterparts living on the same system: some portable programs still ends up writing log files and junk to some user locations. I can stop this by adding pseudo-user credentials and path redirections to Juliawin, but that will cause quite some unexpected behavior:
https://github.com/heetbeet/juliawin/issues/28 . Eg. your `%Downloads%` folder would be somewhere else from Juliawin's perspective.

**Comment by SteffenPL** [comment]
Hi, 

sorry for the late reply. The overview is great, maybe it could be copied to the projects wiki or something 👍 

I played a bit around with it.

On my machine, I ran into this Julia 1.5.3 bug: https://github.com/JuliaLang/julia/issues/38411
(But I guess the best fix is to wait for 1.6)

Is it correct that I either need to rename julia.exe or I need to setup the Julia path in Atom myself?
(I saw in the .bat files the comment that Atom doesn't like julia.exe next to it...)


By the way, feel free to close this ticket anytime!

**Comment by heetbeet** [comment]
Hi, there is a way to manually upgrade to Julia 1.6 (I think): try downloading the 1.6 zip, and replace "packages/julia" with the extracted zip content (keep the same file structure "packages/julia"). You might have to reinstall some of the Julia libraries.

The comment in the bat file is placed at the point where the problem is fixed. The bat file check whether that exe is in the current working directory, then it changes working directories if that is the case. So it should be no worry to the user.
